[quant][graphmode][refactor] Globally record observer nodes

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30553 [quant][graphmode] `insert_quant_dequant` pass support shared class types
* #30552 [quant][graphmode][refactor] Add registerQParams function
* #30551 [quant][graphmode] Insert GetAttr for quantization parameters instead of Constant
* #30550 [quant][graphmode] Make `InsertQuantDeQuantHelper` global
* #30549 [quant][graphmode] Rename QuantizeHelper to InsertQuantDeQuantHelper
* #30548 [quant][graphmode] InsertObservers for shared class types
* **#30547 [quant][graphmode][refactor] Globally record observer nodes**

Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D18784752](https://our.internmc.facebook.com/intern/diff/D18784752)

jerryzh168 committed:
[quant][graphmode][refactor] Globally record observer nodes

Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -454,11 +457,6 @@ void InsertObserversHelper::insertObservers(
-      // Skip observer nodes
-      if (observer_for_input.count(n) != 0) {
-        continue;
-      }
```
Nit: It sill might be worth it not ti add such nodes to `values_to_observe`.

jerryzh168 replied:
would this be a bit redundant though, maybe I can remove the insertObserverFor then?

ZolotukhinM reviewed (approved):
Looks good!

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Globally record observer nodes"


Summary:
att

Test Plan:
test_jit.py test_quantization.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
